packages/doc/src/CommentStore.ts

```diff
@@ -0,0 +1,63 @@
+import { Config, EntityDbMetadata } from "joist-codegen";
+
+/**
+ * Not used within MarkdownCommentStore, but allows additional customisation
+ * of the Field comments by presenting the target place it will be used.
+ */
+export type FieldSourceType = "get" | "set" | "opts";
+
+export type EnumData = { name: string };
+
+/**
+ * The base API for a providing comments to be applied to the various files.
+ *
+ * Focused around Joist concepts (Entities, Fields etc) over the files themselves.
+ */
+export abstract class CommentStore {
```

Comment:
Given we have no shared/base impls, wdyt of making this an `interface`?

Reply:
The only thing we can do with an interface is define the constructor signature right? Can change if that's a willing trade off.

Reply:
Ah yeah, that's true; I'd assumed plugins would want the freedom of "just implement this interface", but if they extend the class, you're right it does document the constructor for things like accepting `config`. Otherwise accepting the `config` would have to be like an `init` method or implicit. Sounds good to keep it then!